client/state/data-layer/wpcom/auth/send-login-email/index.js

```diff
@@ -7,20 +7,43 @@ import { translate } from 'i18n-calypso';
  * Internal dependencies
  */
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { MOBILE_APPS_LOGIN_EMAIL_SEND } from 'state/action-types';
+import {
+	MOBILE_APPS_LOGIN_EMAIL_SEND,
+	MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_FETCH,
+	MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_SUCCESS,
+	MAGIC_LOGIN_SHOW_CHECK_YOUR_EMAIL_PAGE,
+	MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_ERROR,
+} from 'state/action-types';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { registerHandlers } from 'state/data-layer/handler-registry';
 import { infoNotice, errorNotice, successNotice, removeNotice } from 'state/notices/actions';
+import { recordTracksEventWithClientId } from 'state/analytics/actions';
 import config from 'config';
 
 export const sendLoginEmail = action => {
-	const { email, lang_id, locale, redirect_to, showGlobalNotices, actionsOnAPIFetch = [] } = action;
+	const {
+		email,
+		lang_id,
+		locale,
+		redirect_to,
+		showGlobalNotices,
+		loginFormFlow,
+		requestLoginEmailFormFlow,
+	} = action;
 	const noticeAction = showGlobalNotices
 		? infoNotice( translate( 'Sending email' ), { duration: 4000 } )
 		: null;
 	return [
 		...( showGlobalNotices ? [ noticeAction ] : [] ),
-		...actionsOnAPIFetch,
+		...( loginFormFlow || requestLoginEmailFormFlow
+			? [ { type: MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_FETCH } ]
+			: [] ),
+		...( requestLoginEmailFormFlow
+			? [ recordTracksEventWithClientId( 'calypso_login_email_link_submit' ) ]
+			: [] ),
+		...( loginFormFlow
+			? [ recordTracksEventWithClientId( 'calypso_login_block_login_form_send_magic_link' ) ]
+			: [] ),
```

Comment:
😄 

I know I know I know, but still, I approve these changes. thanks for making a hard call

Reply:
I do hope folks know we can also add add tracks and other types of analytics events in the API in server code :D I think it's a lot less interesting for the client to track and non-user initiated actions.